--- xe-guest-utilities/usr/sbin/xe-linux-distribution	2016-05-05 20:31:35.000000000 +0300
+++ xe-guest-utilities-gentoo/usr/sbin/xe-linux-distribution	2016-10-21 17:03:53.881736927 +0300
@@ -293,6 +293,30 @@
 
 }
 
+identify_gentoo()
+{
+    gentoo_release="$1"
+    
+    if [ ! -e "${gentoo_release}" ] ; then
+        return 1
+    fi
+    
+    distro="gentoo"
+    eval $(cat ${gentoo_release} | awk '{ print "release=" $5 }' )
+    
+    if [ -z "${release}" ] ; then
+        return 1
+    fi
+    
+    eval $(echo $release | awk -F. -- '{ print "major=" $1 ; print "minor=" $2 }' )
+    
+    if [ -z "${major}" -o -z "$minor" ] ; then
+        return 1
+    fi
+
+    write_to_output "${distro}" "${major}" "${minor}" "${description}"
+}
+
 if [ $# -eq 1 ] ; then
     exec 1>"$1"
 fi
@@ -306,6 +330,7 @@
     identify_lsb    lsb_release         && exit 0
     identify_debian /etc/debian_version && exit 0
     identify_boot2docker /etc/boot2docker && exit 0
+    identify_gentoo /etc/gentoo-release && exit 0
 
     if [ $# -eq 1 ] ; then
 	rm -f "$1"
