From ba5c850b5968c939d2057d5810bfe93a210b328e Mon Sep 17 00:00:00 2001
From: hhony <hans@knithealth.com>
Date: Wed, 7 Sep 2016 17:22:50 -0700
Subject: [PATCH] CMakeLists: avoid LD errors library should be compiled as
 position independent code

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 02ea263..ab3a74f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -11,7 +11,7 @@ set(CMAKE_CXX_STANDARD_REQUIRED ON)
 add_library(json11 json11.cpp)
 target_include_directories(json11 PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
 target_compile_options(json11
-  PRIVATE -fno-rtti -fno-exceptions -Wall -Wextra -Werror)
+  PRIVATE -fPIC -fno-rtti -fno-exceptions -Wall -Wextra -Werror)
 configure_file("json11.pc.in" "json11.pc" @ONLY)
 
 if (JSON11_BUILD_TESTS)
