diff --git a/hphp/runtime/base/timezone.cpp b/hphp/runtime/base/timezone.cpp
index 1f4004f..3ada227 100644
--- a/hphp/runtime/base/timezone.cpp
+++ b/hphp/runtime/base/timezone.cpp
@@ -27,6 +27,7 @@
 
 #include "hphp/util/functional.h"
 #include "hphp/util/logger.h"
+#include "hphp/util/lock.h"
 #include "hphp/util/text-util.h"
 
 namespace HPHP {
@@ -68,6 +69,8 @@ public:
   }
 };
 static GuessedTimeZone s_guessed_timezone;
+static Mutex s_tzdb_mutex;
+static std::atomic<const timelib_tzdb*> s_tzdb_cache { nullptr };
 
 ///////////////////////////////////////////////////////////////////////////////
 // statics
@@ -98,10 +101,18 @@ void timezone_init() {
   s_tzCache = TimeZoneCache::create(kMaxTimeZoneCache).release();
 }
 
+const timelib_tzdb* timezone_get_builtin_tzdb() {
+  if (s_tzdb_cache != nullptr) return s_tzdb_cache;
+
+  Lock tzdbLock(s_tzdb_mutex);
+  if (s_tzdb_cache == nullptr) s_tzdb_cache = timelib_builtin_db();
+  return s_tzdb_cache;
+}
+
 const timelib_tzdb *TimeZone::GetDatabase() {
   const timelib_tzdb *&Database = s_timezone_data->Database;
   if (Database == nullptr) {
-    Database = timelib_builtin_db();
+    Database = timezone_get_builtin_tzdb();
   }
   return Database;
 }
@@ -182,7 +193,7 @@ bool TimeZone::SetCurrent(const String& zone) {
 }
 
 Array TimeZone::GetNamesToCountryCodes() {
-  const timelib_tzdb *tzdb = timelib_builtin_db();
+  const timelib_tzdb *tzdb = timezone_get_builtin_tzdb();
   int item_count = tzdb->index_size;
   const timelib_tzdb_index_entry *table = tzdb->index;
 
diff --git a/hphp/runtime/base/timezone.h b/hphp/runtime/base/timezone.h
index 1d72eb7..b3a2d3e 100644
--- a/hphp/runtime/base/timezone.h
+++ b/hphp/runtime/base/timezone.h
@@ -153,6 +153,7 @@ private:
 ///////////////////////////////////////////////////////////////////////////////
 
 void timezone_init();
+const timelib_tzdb* timezone_get_builtin_tzdb();
 
 ///////////////////////////////////////////////////////////////////////////////
 }
