diff --git a/hphp/runtime/ext/gd/libgd/gdft.cpp b/hphp/runtime/ext/gd/libgd/gdft.cpp
index 6e7d609..adc2a82 100644
--- a/hphp/runtime/ext/gd/libgd/gdft.cpp
+++ b/hphp/runtime/ext/gd/libgd/gdft.cpp
@@ -62,11 +62,11 @@ gdImageStringFT (gdImage * im, int *brect, int fg, char *fontlist,
 
 #include "gdcache.h"
 
-#ifndef HAVE_FT2BUILD
-# include <freetype/config/ftheader.h>
-#else
+//#ifndef HAVE_FT2BUILD
+//# include <freetype/config/ftheader.h>
+//#else
 # include <ft2build.h>
-#endif
+//#endif
 
 #include FT_FREETYPE_H
 #include FT_GLYPH_H

