diff --git a/CMake/HPHPFindLibs.cmake b/CMake/HPHPFindLibs.cmake
index 60c967b359..f503f99464 100644
--- a/CMake/HPHPFindLibs.cmake
+++ b/CMake/HPHPFindLibs.cmake
@@ -590,8 +590,7 @@ int main() {

   if(NOT "${NOT_REQUIRE_ATOMIC_LINKER_FLAG}")
       message(STATUS "-latomic is required to link hhvm")
-      find_library(ATOMIC_LIBRARY NAMES atomic libatomic.so.1)
-      target_link_libraries(${target} ${ATOMIC_LIBRARY})
+      target_link_libraries(${target} -latomic)
   endif()
   set(CMAKE_REQUIRED_FLAGS ${OLD_CMAKE_REQUIRED_FLAGS})
 endmacro()
